packages/react/src/cx/index.ts

```diff
@@ -0,0 +1,41 @@
+import type { XCSSProp } from '../xcss/xcss-prop';
+
+/**
+ * __cx__
+ *
+ * This function is a utlitiy used to compose xcss fns together.
+ * There is no runtime, it is fully extracted at compile time.
+ *
+ * @example
+ * ```tsx
+ * const style = cx(
+ *   xcss({
+ *     color: 'red',
+ *   }),
+ *   xcss({
+ *     '&:active': {
+ *        color: 'red',
+ *     }
+ *   })
+ * )
+ * ```
+ *
+ * Which would be equivalent to:
+ *
+ * ```tsx
+ * const style = xcss({
+ *   color: 'red',
+ *   '&:active': {
+ *     color: 'red',
+ *   }
+ * })
+ * ```
+ */
+const cx = <Collection extends [...XCSSProp<any, any>[]]>(
```

Comment:
We use a function `cx` instead of a plain array to join overrides. I guess it's due to some type constraints?

Reply:
Yes exactly. By default TypeScript will treat it as a union array, which isn't what we want. We want it to be treated as a const tuple (instead of an array).

Now we could leave it as a tuple, but that means `xcss` prop would allow arrays to be defined, which introduces risk into the system that we'd need to block by lint rules. Instead, we make `xcss` prop only ever take an object, and the `cx` function reduces style args down to a union of object styles.

See: [TypeScript playground](https://www.typescriptlang.org/play?#code/MYewdgzgLgBNCeAbApgFWQD1gXhgbxlERACcAuGAchOQBNKYBfAbgChRJYEUAhAcxi4CAIwCGwANZ8SIAK5haAYRDFyVYYlnIGLVu3DQ4UJMgiCYAbW5pMUADRGT-ALpsY7jzAD0XmAD0Afj0OQ2AMcwAeAEEYW2QFMwsAOhTRMHgLZ2cAPgAKFKTgCAgKKIBKUoswWQBbYWQSZ0Fs-FZ3KAALGQB3GDBkXoBREhkSXIAiAEswADdREkm0qHGytl19TkcUCAAxSYw6czDc63QsB2t+VbbPdx9-AKA)

Reply:
Aha, thanks for explaining this.

Reply:
There already exists the `cx` function internally - would make sense to have it be updated with these types.

So then usage would look like this in some form:

```
import { cssMap, cx } from '@compiled/react';

const styles = cssMap({
  bg: { color: 'red' }
  selected: { textTransform: 'uppercase' }
});

<Component xcss={cx(styles.bg, selected && styles.selected)} />
```

Might want to bikeshed the API though if everyone is using it directly for more use cases we could expose it as a property under `css` to streamline usage? 

```
import { css } from '@compiled/react';

const singleStyle = css({ color: 'red' });

const styles = css.create({
  bg: { color: 'red' }
  selected: { textTransform: 'uppercase' }
});

<Component xcss={css.cx(styles.bg, selected && styles.selected)} />
```

🤔 

@liamqma @JakeLane WDYT?

Reply:
I had a look for the `cx` function and I'm not entirely sure what you're referring to @itsdouges?

Reply:
Ah I think I'm talking about ax...

Reply:
I don't think `ax` is quite the same either it takes an array unless you're saying to adjust the signature and and port any internal existing usage?

Reply:
Iirc it's exported so introducing another function that is very similar in capability isn't productive. We should have an answer to consolidate the two sometime down the line (doesn't have to be now).

Reply:
Sorry, I don't follow you @itsdouges Maybe a quick huddle?

Reply:
`cx` is a new function that hasn't been implemented yet right? I couldn't find any existing usages of `cx` in Compiled or ADS, but I'm not really familiar with the latter

Merging `cx` and `ax` together later down the line sounds ok to me though